module-platform/linux/tests/gendisktestimg.sh

```diff
@@ -1,25 +1,31 @@
-#!/bin/bash -e
-# Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
+#!/bin/bash
+# Copyright (c) 2017-2022, Mudita Sp. z.o.o. All rights reserved.
 # For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
 
-#Copyright (c) 2017-2021, Mudita Sp. z.o.o. All rights reserved.
-#For licensing, see https://github.com/mudita/MuditaOS/LICENSE.md
+set -uo pipefail
 
 if [ $# -ne 1 ]; then
 	echo "Error! Invalid argument count"
-	exit -1
+	exit 1
 fi
+
+L_GIT_DIR=$(git rev-parse --show-toplevel)
+if [ ! -d "$L_GIT_DIR" ]; then L_GIT_DIR=$(readlink -f "$(dirname "$0")"/../../..); fi
```

Comment:
can you please just use the L_GIT_DIR or fail? 
what and why do you want to accomplish here?

Reply:
If you execute a script that relies upon `L_GIT_DIR` then it will currently fail if you run that script using `sudo` because `git` will spit out that you don't own the git repo.  Furthermore, if you download the source instead of cloning the git repo then it will also fail.  What I'm doing is providing a last ditch effort to not fail.